Adding pyproject.toml to make repo installable

DanielYakubov opened the pull request.

DanielYakubov committed:
added pyproject.toml to make repo installable

DanielYakubov committed:
file value was wrong dtype in pyproject.toml

DanielYakubov committed:
toml name issue fix

DanielYakubov committed:
wrong name on toml, again

DanielYakubov committed:
added __init__.py, required for packaging

DanielYakubov commented on pyproject.toml:
```diff
@@ -0,0 +1,17 @@
+name = "UD-UM"
+version = "1.0.0"
+authors = [
+    {name = "Arya McCarthy", email = "placeholder@gmail.com"},
```
this email needs to be filled out, sorry, I could not find one online.

kylebgorman commented on pyproject.toml:
```diff
@@ -0,0 +1,17 @@
+authors = [
+    {name = "Arya McCarthy", email = "placeholder@gmail.com"},
+]
+description = "Python package developer's cheat sheet (using pyproject.toml)."
```
fill in

DanielYakubov replied:
embarrassing. thanks.

kylebgorman commented on pyproject.toml:
```diff
@@ -0,0 +1,17 @@
+requires = ["flit_core >=3.8.0,<4"]
+
+[project]
+name = "UD-UM"
```
not a great name for this. I would probably go with Arya's original `ud-compatibility`.

DanielYakubov replied:
it has to do with the folder name - I think I would have to rename the `ud-um` folder to `ud-compatibility` to do this - https://stackoverflow.com/questions/69230525/why-does-poetry-build-raise-moduleorpackagenotfound-exception

kylebgorman replied:
This is of course up to @aryamccarthy but I think then that directory should also be renamed, since that's also a little terse.

kylebgorman reviewed (commented):
I put a few notes here

DanielYakubov committed:
updated pyproject.toml - left a copy paste error and changed name to lowercase

DanielYakubov committed:
changed version

DanielYakubov committed:
added email

DanielYakubov committed:
added maintainer; switched name back to UM-UD, required for build

DanielYakubov commented on pyproject.toml:
```diff
@@ -0,0 +1,20 @@
+requires = ["flit_core >=3.8.0,<4"]
+
+[project]
+name = "UD_UM"
```
@aryamccarthy to rename this to ud-compatibility (and thus name the package as such), we need to change the UD_UM folder to ud-compatibility. Is that okay with you?

aryamccarthy committed:
Move UD_UM to ud-compatibility

This is a move to match the package name to the repo name, while sticking to nice naming conventions.

aryamccarthy commented:
Yes, renamed and merged.

DanielYakubov commented:
@aryamccarthy ty, is it alright if I upload this to Pypi?

aryamccarthy commented:
Go for it.

DanielYakubov commented:
@aryamccarthy Sorry, my mistake. The inner dir needs to be `ud_compatibility` (not with a dash, but an underscore) otherwise generating the .whl fails with flit. I hope you don't mind making that small change, otherwise, I will open a new PR with this change.

from the docs - "The name your package will have on PyPI. This field is required. For Flit, this name, with any hyphens replaced by underscores, is also the default value of the import name"  - https://flit.pypa.io/en/stable/pyproject_toml.html
